samcli/local/docker/container.py

```diff
@@ -334,32 +334,30 @@ def wait_for_logs(self, stdout=None, stderr=None):
 
         self._write_container_output(logs_itr, stdout=stdout, stderr=stderr)
 
-    def wait_for_socket_connection(self):
+    def _wait_for_socket_connection(self) -> None:
         """
         Waits for a successful connection to the socket used to communicate with Docker.
         """
-        start_time = time.time()
-        sleep = 0.1
-        while True:
+
+        def can_connect_to_socket() -> bool:
```

Comment:
nit: this function could be defined at class level with an underscore in the beginning. The more we add inline functions the harder it is to understand what is going on.

Reply:
done 👍